# 
# these changes are necessary to allow keycloak to work behind a reverse proxy
#
# see: https://github.com/keycloak/keycloak-documentation/blob/master/server_installation/topics/clustering/load-balancer.adoc
#
--- opt/keycloak/standalone/configuration/standalone-ha.xml	2017-05-03 08:10:23.000000000 +0200
+++ opt/keycloak/standalone/configuration/standalone-ha.xml	2017-08-03 11:25:49.000000000 +0200
@@ -408,7 +408,8 @@
             <buffer-cache name="default"/>
             <server name="default-server">
                 <ajp-listener name="ajp" socket-binding="ajp"/>
-                <http-listener name="default" socket-binding="http" redirect-socket="https"/>
+                <http-listener name="default" socket-binding="http" redirect-socket="proxy-https"
+                           proxy-address-forwarding="true"/>
                 <host name="default-host" alias="localhost">
                     <location name="/" handler="welcome-content"/>
                     <filter-ref name="server-header"/>
@@ -521,5 +521,6 @@
         <outbound-socket-binding name="mail-smtp">
             <remote-destination host="localhost" port="25"/>
         </outbound-socket-binding>
+        <socket-binding name="proxy-https" port="443"/>
     </socket-binding-group>
 </server>
--- opt/keycloak/standalone/configuration/standalone.xml	2017-05-03 08:10:23.000000000 +0200
+++ opt/keycloak/standalone/configuration/standalone.xml	2017-08-03 11:25:49.000000000 +0200
@@ -369,7 +369,8 @@
         <subsystem xmlns="urn:jboss:domain:undertow:3.0">
             <buffer-cache name="default"/>
             <server name="default-server">
-                <http-listener name="default" socket-binding="http" redirect-socket="https"/>
+                <http-listener name="default" socket-binding="http" redirect-socket="proxy-https"
+                           proxy-address-forwarding="true"/>
                 <host name="default-host" alias="localhost">
                     <location name="/" handler="welcome-content"/>
                     <filter-ref name="server-header"/>
@@ -473,5 +473,6 @@
         <outbound-socket-binding name="mail-smtp">
             <remote-destination host="localhost" port="25"/>
         </outbound-socket-binding>
+        <socket-binding name="proxy-https" port="443"/>
     </socket-binding-group>
 </server>
